File: Assets/scenes/landinglaunchscene/script/GravityLight.js
#pragma strict

var rotationSpeed : float = 0.0f;

var startOn : boolean = false;

private var on : boolean = false;
private var lights : Light[];

function Start () {
	lights = GetComponentsInChildren.<Light>();
	setState(startOn);
}

function FixedUpdate(){
	if(on){
		transform.localRotation *= Quaternion.Euler(rotationSpeed,0,0);
	}
}


function setState( state : boolean){

	on = state;
	if(on){
		for (var l : Light in lights){
			l.color = Color(1,0,0);
		}
	} else {
		for (var l : Light in lights){
			l.color = Color(0,0,0);
		}
	}

}

File: Assets/scenes/landinglaunchscene/script/GravityLight.cs
using UnityEngine;
using System;



public class GravityLight:MonoBehaviour{
	
	public float rotationSpeed = 0.0f;
	
	public bool startOn = false;
	
	bool on = false;
	Light[] lights;
	
	public void Start() {
		lights = GetComponentsInChildren<Light>();
		setState(startOn);
	}
	
	public void FixedUpdate(){
		if(on){
			transform.localRotation *= Quaternion.Euler(rotationSpeed,0.0f,0.0f);
		}
	}
	
	
	public void setState(bool state){
	
		on = state;
		if(on){
			foreach(Light l in lights){
				l.color = new Color(1.0f,0.0f,0.0f);
			}
		} else {
			foreach(Light l in lights){
				l.color = new Color(0.0f,0.0f,0.0f);
			}
		}
	
	}
}